contribs/gmf/src/directives/authenticationdirective.js

```diff
@@ -0,0 +1,140 @@
+goog.provide('gmf.AuthenticationController');
+goog.provide('gmf.authenticationDirective');
+
+goog.require('gmf');
+goog.require('gmf.Authentication');
+goog.require('ngeo.modalDirective');
+
+
+gmfModule.value('gmfAuthenticationTemplateUrl',
+    /**
+     * @param {angular.JQLite} element Element.
+     * @param {angular.Attributes} attrs Attributes.
+     */
+    function(element, attrs) {
+      var templateUrl = attrs['gmfAuthenticationTemplateurl'];
+      return templateUrl !== undefined ? templateUrl :
+          gmf.baseTemplateUrl + '/authentication.html';
+    });
+
+
+/**
+ * An "authentication" directive for a GeoMapFish application.
+ *
+ * @example
+ * <gmf-authentication></gmf-authentication>
+ *
+ * @param {string} gmfAuthenticationTemplateUrl Url to template.
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfAuthentication
+ */
+gmf.authenticationDirective = function(gmfAuthenticationTemplateUrl) {
+  return {
+    scope: {},
+    controller: 'GmfAuthenticationController',
+    controllerAs: 'authCtrl',
+    templateUrl: gmfAuthenticationTemplateUrl
+  };
+};
+
+gmfModule.directive('gmfAuthentication', gmf.authenticationDirective);
+
+
+
+/**
+ * @param {angular.Scope} $scope The directive's scope.
+ * @param {gmf.Authentication} gmfAuthentication GMF Authentication service
+ * @constructor
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfAuthenticationController
+ */
+gmf.AuthenticationController = function($scope, gmfAuthentication) {
+
+  /**
+   * @type {gmf.Authentication}
+   * @export
+   */
+  this.gmfAuthentication = gmfAuthentication;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.resetPasswordModalShown = false;
+
+  /**
+   * @type {string}
+   * @export
+   */
+  this.loginVal = '';
+
+  /**
+   * @type {string}
+   * @export
+   */
+  this.pwdVal = '';
+
+};
+
+
+/**
+ * @export
+ */
+gmf.AuthenticationController.prototype.login = function() {
+  this.gmfAuthentication.login(this.loginVal, this.pwdVal).then(goog.bind(
+      /**
+       * @param {gmf.AuthenticationLoginResponse} respData
+       */
+      function(respData) {
+        if (!respData.status) {
```

Comment:
I'm not sure what's you're doing here.
Can you get a 200 but not being logged ?
It seems the server return a 401.

Plug it to the failure callback then.